### PR TITLE
added a uniform distribution over a collection of objects

### PR DESCRIPTION
**Pull request by zsunberg**
One question is naming. This clashes with `Distributions.Uniform`. Another option would be `UniformCat`. I think `Uniform` is better, but could be persuaded otherwise.

### CODE REVIEW

File: src/distributions/uniform.jl [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+rand(rng::AbstractRNG, d::Uniform) = rand(rng, d.collection)
+rand(rng::AbstractRNG, d::Uniform{<:NamedTuple}) = d.collection[rand(rng, 1:length(d.collection))]
+
+support(d::Uniform) = d.collection
+sampletype(::Type{Uniform{C, T}}) where {C,T} = T
+
+function pdf(d::Uniform, s)
+    d._set = something(d._set, Set(d.collection))
+    if s in d._set
+        return 1/length(d.collection)
```
Comment (rejuvyesh) [inline src/distributions/uniform.jl]: `1.0/length(d.collection)`

File: src/distributions/uniform.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+"""
+mutable struct Uniform{C, T}
+    collection::C
+    _set::Union{Set{T}, Nothing} # keep track of what's in the collection to make pdf more efficient
+end
+
+Uniform(c) = Uniform{typeof(c), eltype(c)}(c, nothing)
+Uniform(c::Set) = Uniform{typeof(c), eltype(c)}(c, c)
+
+rand(rng::AbstractRNG, d::Uniform) = rand(rng, d.collection)
```
Comment (rejuvyesh) [inline src/distributions/uniform.jl]: Unclear about the rationale here. If you are allowing for non-sets and converting it into sets and giving them equal weights, then how can you sample from the collection which might have multiple instances of the same and therefore higher weight?

Comment (zsunberg) [reply src/distributions/uniform.jl]: Yeah, thanks, I need to redesign this.

**Comment by zsunberg** [comment]
@rejuvyesh do you think it's good now?

File: src/distributions/uniform.jl [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return 1.0/length(d.set)
+    else
+        return 0.0
+    end
+end
+
+# mean(d::Uniform) = mean(d.set)
+# mode(d::Uniform) = mode(d.set)
+# 
+# function weighted_iterator(d::Uniform)
```
Comment (rejuvyesh) [inline src/distributions/uniform.jl]: Delete these?

Comment (zsunberg) [reply src/distributions/uniform.jl]: oh yeah :man_facepalming:

**Pull request merged**